homeassistant/components/weather/__init__.py

```diff
@@ -173,46 +419,108 @@ def ozone(self) -> float | None:
+
+        if (native_visibility_unit := self.native_visibility_unit) is None:
+            return None
+
+        if weather_option_visibility_uom := self._weather_option_visibility_uom:
+            return weather_option_visibility_uom
+
+        return native_visibility_unit
 
     @property
     def forecast(self) -> list[Forecast] | None:
         """Return the forecast in native units."""
         return self._attr_forecast
 
     @property
-    def precipitation_unit(self) -> str | None:
+    def native_precipitation_unit(self) -> str | None:
         """Return the native unit of measurement for accumulated precipitation."""
-        return self._attr_precipitation_unit
+        return self._attr_native_precipitation_unit
+
+    @property
+    def precipitation_unit(self) -> str | None:
+        """Return the converted unit of measurement for precipitation."""
+        if (precipitation_unit := self._attr_precipitation_unit) is not None:
+            return precipitation_unit
+
+        if (native_precipitation_unit := self.native_precipitation_unit) is None:
+            return None
+
+        if _weather_option_precipitation_uom := self._weather_option_precipitation_uom:
+            return _weather_option_precipitation_uom
+
+        return native_precipitation_unit
 
     @property
     def precision(self) -> float:
         """Return the precision of the temperature value, after unit conversion."""
         if hasattr(self, "_attr_precision"):
             return self._attr_precision
+        if self._override:
+            return (
+                PRECISION_TENTHS
+                if self.hass.config.units.temperature_unit == TEMP_CELSIUS
+                else PRECISION_WHOLE
+            )
         return (
             PRECISION_TENTHS
-            if self.hass.config.units.temperature_unit == TEMP_CELSIUS
+            if self.temperature_unit == TEMP_CELSIUS
             else PRECISION_WHOLE
         )
 
     @final
     @property
-    def state_attributes(self):
+    def state_attributes(self):  # noqa: C901
         """Return the state attributes, converted from native units to user-configured units."""
         data = {}
-        if self.temperature is not None:
+
+        precision = self.precision
+
+        if (temperature := self.temperature) is not None and (
+            temp_unit := self.temperature_unit
+        ) is not None:
             data[ATTR_WEATHER_TEMPERATURE] = show_temp(
                 self.hass,
-                self.temperature,
-                self.temperature_unit,
-                self.precision,
+                temperature,
+                temp_unit,
+                precision,
+            )
+        elif (
+            (temperature := self.native_temperature) is not None
+            and (native_temp_unit := self.native_temperature_unit) is not None
+            and (temp_unit := self.temperature_unit) is not None
+        ):
+            with suppress(ValueError):
```

Comment:
Either by someone setting a uom which is not valid or in this case a temperature comes in which can not converted into a float.
I think it would be good if it continues instead of breaking (and thereby skipping this attribute until proper values are coming in).